org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/basic/table/internal/InternalTableRow.java

```diff
@@ -207,15 +209,16 @@ public Object getCellValue(int columnIndex) {
 
   @Override
   public List<Object> getKeyValues() {
-    List<Object> pk = new ArrayList<>();
-    if (getTable() != null) {
-      int[] keyColumns = getTable().getColumnSet().getKeyColumnIndexes();
-      if (keyColumns.length == 0) {
-        keyColumns = getTable().getColumnSet().getAllColumnIndexes();
-      }
-      for (int keyIndex : keyColumns) {
-        pk.add(getCell(keyIndex).getValue());
-      }
+    if (getTable() == null) {
+      return new ArrayList<>(0);
+    }
+    int[] keyColumns = getTable().getColumnSet().getKeyColumnIndexes();
+    if (keyColumns.length == 0) {
+      keyColumns = getTable().getColumnSet().getAllColumnIndexes();
+    }
```

Comment:
This would save creating an empty array:
```suggestion
    ColumnSet columnSet = getTable().getColumnSet();
    int[] keyColumns = columnSet.getKeyColumnCount() > 0 ? columnSet.getKeyColumnIndexes() : columnSet.getAllColumnIndexes();
```